command/crypto/otp/verify.go

```diff
@@ -67,23 +73,73 @@ as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms"
 }
 
 func verifyAction(ctx *cli.Context) error {
-	args := ctx.Args()
-	secretFile := args.Get(0)
+	var (
+		secret string
+		// defaults
+		period = uint(30)
+		digits = 6
+		algStr = "SHA1"
+	)
 
+	secretFile := ctx.String("secret")
+	if len(secretFile) == 0 {
+		args := ctx.Args()
+		if len(args) == 0 {
+			return errs.RequiredFlag(ctx, "secret")
+		}
+		secretFile = args[0]
+	}
 	b, err := ioutil.ReadFile(secretFile)
 	if err != nil {
 		return errs.FileError(err, secretFile)
 	}
-	secret := string(b)
+	secret = string(b)
 	if strings.HasPrefix(secret, "otpauth://") {
-		url, err := otp.NewKeyFromURL(secret)
+		otpKey, err := otp.NewKeyFromURL(secret)
+		if err != nil {
+			return errors.Wrap(err, "error parsing TOTP key from URL")
+		}
+
+		u, err := url.Parse(strings.TrimSpace(secret))
 		if err != nil {
-			return err
+			return errors.Wrap(err, "error parsing TOTP Key URI in secret file")
+		}
+		q := u.Query()
+
+		secret = otpKey.Secret()
+		// period query param
+		if periodStr := q.Get("period"); len(periodStr) > 0 {
+			period64, err := strconv.ParseUint(periodStr, 10, 64)
```

Comment:
To avoid overflows converting from uint64/int64 to uint/int, I would replace the 64 to `strconv.IntSize`. Same a few lines below.

Reply:
Use bitSize `0`. See https://golang.org/pkg/strconv/#ParseInt
```
The bitSize argument specifies the integer type that the result
must fit into. Bit sizes 0, 8, 16, 32, and 64 correspond to int, 
int8, int16, int32, and int64. If bitSize is below 0 or above 64, 
an error is returned.
```